backend/internal/storage/postgres/rubrics.go

```diff
@@ -0,0 +1,91 @@
+package postgres
+
+import (
+	"context"
+
+	"github.com/CamPlume1/khoury-classroom/internal/errs"
+	"github.com/CamPlume1/khoury-classroom/internal/models"
+	"github.com/jackc/pgx/v5"
+)
+
+func (db *DB) CreateRubric(ctx context.Context, rubricData models.Rubric) (models.Rubric, error) {
+    err := db.connPool.QueryRow(ctx, "INSERT INTO rubrics (name, org_id, classroom_id, reusable) VALUES ($1, $2, $3, $4) RETURNING *",
+        rubricData.Name,
+        rubricData.OrgID,
+        rubricData.ClassroomID,
+        rubricData.Reusable).Scan(
+        &rubricData.ID,
+        &rubricData.Name,
+        &rubricData.OrgID,
+        &rubricData.ClassroomID,
+        &rubricData.Reusable,
+        &rubricData.CreatedAt)
+    if err != nil {
+        return models.Rubric{}, errs.NewDBError(err)
+    }
+
+    return rubricData, nil
+}
+
+func (db *DB) AddItemToRubric(ctx context.Context, rubricItemData models.RubricItem) (models.RubricItem, error) {
+    err := db.connPool.QueryRow(ctx, "INSERT INTO rubric_items (rubric_id, point_value, explanation) VALUES ($1, $2, $3) RETURNING *",
+        rubricItemData.RubricID,
+        rubricItemData.PointValue,
+        rubricItemData.Explanation).Scan(
+        &rubricItemData.ID,
+        &rubricItemData.RubricID,
+        &rubricItemData.PointValue,
+        &rubricItemData.Explanation,
+        &rubricItemData.CreatedAt)
+
+    if err != nil {
+        return models.RubricItem{}, errs.NewDBError(err)
+    }
+
+    return rubricItemData, nil
+}
+
+
+func (db *DB) GetFullRubric(ctx context.Context, rubricID int64) (models.FullRubric, error) {
+    rubric, err := db.GetRubric(ctx, rubricID)
+    if err != nil {
+        return models.FullRubric{}, errs.NewDBError(err)
+    }
+
+    rubricItems, err := db.GetRubricItems(ctx, rubricID)
+    if err != nil {
+        return models.FullRubric{}, errs.NewDBError(err)
+    }
+
+    var fullRubric models.FullRubric
+    fullRubric.Rubric = rubric
+    fullRubric.RubricItems = rubricItems
+
+    return fullRubric, nil
+}
```

Comment:
IMO this should be put in the services layer since how rubrics get constructed from the separate components is business logic

Reply:
good comment